Automattic/wp-calypso
# Social: Show free trial for Advanced plan

This change updates the Lightbox UI to handle the 0$ free trial month for Social

## Proposed Changes
* In a lot of places the discounted price was checked by a boolean check, which I changed to an `isNumber` check because 0 would be falsy
* Updated some time frame messages 

## Testing Instructions

* Check out this PR and Sandbox the store with your wpcom sandbox, by adding `define( 'USE_STORE_SANDBOX', true );` to your 0-sandbox.php file
* Start up Jetpack Cloud then go to the `/pricing` page. You should see the correct messaging on the product page and the Social product card.


![CleanShot 2024-01-11 at 14 37 52 png](https://github.com/Automattic/wp-calypso/assets/36671565/17b8483f-6c7e-4aab-95e5-874b8f82b395)

![CleanShot 2024-01-11 at 14 38 06 png](https://github.com/Automattic/wp-calypso/assets/36671565/06ae8e3a-82de-4518-a233-f7b149b6da29)

## Pre-merge Checklist
